torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -148,7 +148,8 @@ struct TORCH_API AccumulateGrad : public Node {
             new_grad.sizes(),
             new_grad.options()));
       } else {
-        if (new_grad.is_sparse() || new_grad.is_sparse_csr()) {
+        if (new_grad.is_sparse() || new_grad.is_sparse_csr() ||
```

Comment:
Is this also temporary? Can we eventually coerce nested tensor to have the same layout as the variable.

Reply:
Yes I think this is temporary and the above changes I mentioned in grad_layout_contract would remove this check